### PR TITLE
[video_player_platform_interface] Bump version for NNBD stable

### PR DESCRIPTION
**Pull request by ditman**
Includes:

* Remove Mockito, only used for extends/implements tests.
* Migrated tests to null-safety.
  * ~Skipped a couple of tests~ Manually modified a generated pigeon file because of: https://github.com/flutter/flutter/issues/76405

## Pre-launch Checklist

- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: packages/video_player/video_player_platform_interface/CHANGELOG.md [hunk 1]
```diff
@@ -1,27 +1,15 @@
...
-
-* Bump Dart SDK to support null safety.
-
-## 3.0.0-nullsafety.1
-
 * Make DataSource's `uri` parameter nullable.
+* `messages.dart` sets Dart `2.12`.
 
-## 3.0.0-nullsafety
+## 3.0.0
```
Comment (blasten) [inline packages/video_player/video_player_platform_interface/CHANGELOG.md:10]: i'm confused. should this just be 3.0.0 breaking change? did we break a prerelease breaking change? 😅

Comment (ditman) [reply packages/video_player/video_player_platform_interface/CHANGELOG.md:10]: Yeah, `4.0.0-nullsafety.0` was published to pub. I doubt pub will let me go back to 3.0.0 :/

**Review by blasten — APPROVED** [review]
LGTM

**Comment by stuartmorgan** [comment]
Shouldn't generated pigeon code be edited as necessary to migrate it correctly?

**Comment by ditman** [comment]
> Shouldn't generated pigeon code be edited as necessary to migrate it correctly?

@stuartmorgan shouldn't pigeon be fixed so the generated code doesn't need manual intervention?

In this case, it seems it's purely test code that is using the generated Pigeon code "wrong", so that's why I decided to skip the tests. I can tweak the pigeon code for tests to pass, but it seems that more classes that are not being tested (as thoroughly) might have the same problem.

**Comment by stuartmorgan** [comment]
> @stuartmorgan shouldn't pigeon be fixed so the generated code doesn't need manual intervention?

Definitely, but since this PR isn't blocked on that the question is what to do in the short term. If it's trivial to manually fix the generated code for now (which is what it sounded like?) to match what it *should* generate, that seems significantly better than temporarily losing test coverage.

> In this case, it seems it's purely test code that is using the generated Pigeon code "wrong", so that's why I decided to skip the tests. I can tweak the pigeon code for tests to pass, but it seems that more classes that are not being tested (as thoroughly) might have the same problem.

Now I'm confused though. Is the generated code wrong, or are the tests wrong? If the tests are wrong we should fix/remove them, but the from the bug you filed it sounds like it's the generated code.

**Comment by ditman** [comment]
> If it's trivial to manually fix the generated code for now (which is what it sounded like?) to match what it should generate, that seems significantly better than temporarily losing test coverage.

Gotcha, will do this @stuartmorgan.

> Is the generated code wrong, or are the tests wrong? If the tests are wrong we should fix/remove them, but the from the bug you filed it sounds like it's the generated code.

Part of the test code is generated (the [fake VideoPlayerApi](https://github.com/flutter/plugins/blob/master/packages/video_player/video_player_platform_interface/lib/test.dart#L48) used in tests). The problem is that the fake videoplayer API attempts to deserialize a message with nulls in it, and that's the part that explodes.

The codepath in the `test.dart` file does not happen in the real apps I have to test (but maybe in users' apps), and that's why the example app works in sound mode. It's some particular messages that Dart is attempting to deserialize (but in this case, most likely, will never come back from pigeon like that). Also, that message *must* contain certain null values, because they're used in the native side to decide what to do (how to create the player).

**Comment by ditman** [comment]
Also, the fix is coming! Yay! https://github.com/flutter/packages/pull/292

**Comment by ditman** [comment]
Published

**Pull request merged**